geode-for-redis/src/main/java/org/apache/geode/redis/internal/RedisProperties.java

```diff
@@ -16,38 +16,90 @@
 package org.apache.geode.redis.internal;
 
 import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.logging.internal.log4j.api.LogService;
 
 public class RedisProperties {
-  /** System Property Names **/
-  public static final String REDIS_REGION_NAME_PROPERTY = "geode-for-redis-region-name";
-  public static final String WRITE_TIMEOUT_SECONDS = "geode-for-redis-write-timeout-seconds";
-  public static final String EXPIRATION_INTERVAL_SECONDS =
-      "geode-for-redis-expiration-interval-seconds";
+  private static final Logger logger = LogService.getLogger();
+
+  private static final String PREFIX = "geode.geode-for-redis-";
+  public static final String REDIS_REGION_NAME_PROPERTY = PREFIX + "region-name";
+  public static final String WRITE_TIMEOUT_SECONDS = PREFIX + "write-timeout-seconds";
+  public static final String EXPIRATION_INTERVAL_SECONDS = PREFIX + "expiration-interval-seconds";
+  public static final String REGION_BUCKETS = PREFIX + "region-buckets";
+  public static final String UNAUTHENTICATED_MAX_ARRAY_SIZE =
+      PREFIX + "unauthenticated-max-array-size";
+  public static final String UNAUTHENTICATED_MAX_BULK_STRING_LENGTH =
+      PREFIX + "unauthenticated-max-bulk-string-length";
+
+
+  private static String convertToGemfire(String geodeName) {
+    return "gemfire." + geodeName.substring("geode.".length());
+  }
+
+  private static void validatePropertyName(String geodeName) {
+    if (!geodeName.startsWith(PREFIX)) {
+      throw new IllegalStateException("Property names must start with \"" + PREFIX + "\"");
+    }
+  }
 
   public static String getStringSystemProperty(String propName, String defaultValue) {
-    String geodeValue = System.getProperty("geode." + propName, defaultValue);
-    String gemfireValue = System.getProperty("gemfire." + propName, defaultValue);
+    validatePropertyName(propName);
 
+    String geodeValue = System.getProperty(propName, defaultValue);
     if (StringUtils.isNotEmpty(geodeValue) && !geodeValue.equals(defaultValue)) {
```

Comment:
Do we actually need to check if the value here is equal the the default? If the property exists, we should just return it; checking if it's equal to the default just results in extra work and the same outcome in the end. This comment also applies to the check on line 56.

Reply:
Are you suggesting that it should use System.getProperty(propName) instead of System.getProperty(propName, defaultValue)? I think that it is a good idea since it is simpler and since we never end up using geodeValue if it is defaultValue.